engine/execution/state/bootstrap/bootstrap_test.go

```diff
@@ -89,3 +92,104 @@ func TestBootstrapLedger_ZeroTokenSupply(t *testing.T) {
 		}
 	})
 }
+
+// TestBootstrapLedger_TokenTransfer bootstraps a ledger with:
+// - transaction fees
+// - storage fees
+// - minimum account balance
+// - initial token supply
+// Then runs an empty transaction to trigger the bookkeeping parts of a transaction:
+// - payer has balance to cover the transaction fees check
+// - account storage check
+// - transaction fee deduction
+// This tests that the state commitment has not changed for the bookkeeping parts of the transaction.
+func TestBootstrapLedger_TokenTransfer(t *testing.T) {
+	expectedStateCommitmentBytes, _ := hex.DecodeString("df1937164702f557d7457f8fdd6040a81dfbcbdbfb93ec187c3bf1cac414f5c0")
+	expectedStateCommitment, err := flow.ToStateCommitment(expectedStateCommitmentBytes)
+	require.NoError(t, err)
+
+	unittest.RunWithTempDir(t, func(dbDir string) {
+
+		chain := flow.Mainnet.Chain()
+
+		metricsCollector := &metrics.NoopCollector{}
+		wal := &fixtures.NoopWAL{}
+		ls, err := completeLedger.NewLedger(wal, 100, metricsCollector, zerolog.Nop(), completeLedger.DefaultPathFinderVersion)
+		require.NoError(t, err)
+		compactor := fixtures.NewNoopCompactor(ls)
+		<-compactor.Ready()
+		defer func() {
+			<-ls.Done()
+			<-compactor.Done()
+		}()
+
+		stateCommitment, err := NewBootstrapper(zerolog.Nop()).BootstrapLedger(
+			ls,
+			unittest.ServiceAccountPublicKey,
+			chain,
+			fvm.WithInitialTokenSupply(unittest.GenesisTokenSupply),
+			fvm.WithAccountCreationFee(fvm.DefaultAccountCreationFee),
+			fvm.WithMinimumStorageReservation(fvm.DefaultMinimumStorageReservation),
+			fvm.WithTransactionFee(fvm.DefaultTransactionFees),
+			fvm.WithStorageMBPerFLOW(fvm.DefaultStorageMBPerFLOW),
+		)
+		require.NoError(t, err)
+
+		storageSnapshot := state.NewLedgerStorageSnapshot(ls, stateCommitment)
+		vm := fvm.NewVirtualMachine()
+
+		ctx := fvm.NewContext(
+			fvm.WithChain(chain),
+			fvm.WithTransactionFeesEnabled(true),
+			fvm.WithTransactionFeesEnabled(true),
+			fvm.WithAccountStorageLimit(true),
+			fvm.WithSequenceNumberCheckAndIncrementEnabled(false),
+			fvm.WithAuthorizationChecksEnabled(false),
+		)
+
+		sc := systemcontracts.SystemContractsForChain(chain.ChainID())
+
+		txBody := flow.NewTransactionBody().
+			SetScript([]byte(`
+				transaction() {
```

Comment:
This seems to be an empty transaction instead of a token transfer. Did I miss anything?

Reply:
Yes! I did update the comment!